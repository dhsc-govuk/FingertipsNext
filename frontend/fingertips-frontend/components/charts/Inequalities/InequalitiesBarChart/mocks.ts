import { InequalitiesBarChartData } from '@/components/charts/Inequalities/helpers/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaCode: 'A1425',
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: true,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
    },
  },
});
