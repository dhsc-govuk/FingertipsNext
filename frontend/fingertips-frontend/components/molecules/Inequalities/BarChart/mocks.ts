import { InequalitiesBarChartData } from '@/components/organisms/Inequalities/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
    },
  },
});
