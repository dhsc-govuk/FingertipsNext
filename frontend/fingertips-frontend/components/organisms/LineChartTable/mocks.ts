export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'Female',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'Female',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
  {
    ...MOCK_ENGLAND_DATA,
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2004,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
];
