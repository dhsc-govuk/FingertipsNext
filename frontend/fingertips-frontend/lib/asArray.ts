export const asArray = (value?: string | string[]): string[] => {
  if (value) {
    return Array.isArray(value) ? value : [value];
  }
  return [];
};
