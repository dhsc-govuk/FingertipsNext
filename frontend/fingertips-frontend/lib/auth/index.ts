import { AuthConfigFactory } from '@/lib/auth/config';
import NextAuth from 'next-auth';

export const { handlers, signIn, signOut, auth } = NextAuth(
  AuthConfigFactory.getConfig()
);
