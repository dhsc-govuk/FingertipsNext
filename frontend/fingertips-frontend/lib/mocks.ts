import {
  AgeData,
  DeprivationData,
  HealthDataPoint,
  HealthDataPointTrendEnum,
  SexData,
} from '@/generated-sources/ft-api-client';
import { mockDatePeriod } from '@/mock/data/mockDatePeriod';

export const noDeprivation: DeprivationData = {
  sequence: 1,
  type: 'All',
  value: 'All',
  isAggregate: true,
};

export const personsSex: SexData = {
  value: 'Persons',
  isAggregate: true,
};

export const maleSex: SexData = {
  value: 'Male',
  isAggregate: false,
};

export const femaleSex: SexData = {
  value: 'Female',
  isAggregate: false,
};

export const allAgesAge: AgeData = {
  value: 'All ages',
  isAggregate: true,
};

export const disaggregatedAge = (age: string) => {
  return {
    value: age,
    isAggregate: false,
  };
};

export const healthDataPoint: HealthDataPoint = {
  count: 389,
  lowerCi: 441.69151,
  upperCi: 578.32766,
  value: 278.29134,
  datePeriod: mockDatePeriod(2006),
  sex: personsSex,
  ageBand: allAgesAge,
  trend: HealthDataPointTrendEnum.NotYetCalculated,
  deprivation: noDeprivation,
};
