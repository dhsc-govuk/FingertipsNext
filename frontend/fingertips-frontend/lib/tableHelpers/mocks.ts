export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'All',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'All',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Male',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Female',
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Male',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Female',
      },
    ],
  },
];

export const MOCK_PARENT_DATA = {
  areaCode: 'E40000014',
  areaName: 'NHS North West Region',
  healthData: [
    {
      year: 2008,
      count: 777,
      value: 1000,
      lowerCi: 500,
      upperCi: 1500,
      ageBand: 'All',
      sex: 'All',
    },
    {
      year: 2004,
      count: 777,
      value: 1000,
      lowerCi: 500,
      upperCi: 1500,
      ageBand: 'All',
      sex: 'All',
    },
  ],
};
