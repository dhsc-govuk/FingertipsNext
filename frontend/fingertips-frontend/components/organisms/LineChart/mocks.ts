import {
  HealthDataForArea,
  HealthDataPointTrendEnum,
} from '@/generated-sources/ft-api-client';
import { allAgesAge, noDeprivation, personsSex } from '@/lib/mocks';

export const mockIndicatorData: HealthDataForArea[] = [
  {
    areaCode: 'A1425',
    areaName: 'North FooBar',
    healthData: [
      {
        count: 389,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 278.29134,
        year: 2006,
        periodLabel: '2006',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 267,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 703.420759,
        year: 2004,
        periodLabel: '2004',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        count: 654,
        lowerCi: 750.69151,
        upperCi: 800.32766,
        value: 786.27434,
        year: 2010,
        periodLabel: '2010',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 440.69151,
        upperCi: 420.32766,
        value: 435.420759,
        year: 2007,
        periodLabel: '2007',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1427',
    areaName: 'East FooBar',
    healthData: [
      {
        count: 953,
        lowerCi: 460.69151,
        upperCi: 500.32766,
        value: 478.27434,
        year: 2020,
        periodLabel: '2020',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 220.69151,
        upperCi: 250.32766,
        value: 234.420759,
        year: 2012,
        periodLabel: '2012',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
];

export const mockBenchmarkData: HealthDataForArea = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      count: 389,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 278.29134,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 267,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 703.420759,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};

export const mockParentData: HealthDataForArea = {
  areaCode: 'P001',
  areaName: 'Parent',
  healthData: [
    {
      count: 100,
      lowerCi: 200,
      upperCi: 400,
      value: 300,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 101,
      lowerCi: 201,
      upperCi: 401,
      value: 301,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};
