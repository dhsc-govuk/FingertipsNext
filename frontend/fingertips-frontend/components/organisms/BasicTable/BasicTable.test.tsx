import { render, screen } from '@testing-library/react';
import { healthDataPoint } from '@/lib/mocks';
import {
  BasicTableData,
  BasicTable,
} from '@/components/organisms/BasicTable/index';

const mockIndicatorData: BasicTableData[] = [
  {
    indicatorId: 1,
    indicatorName: ' ',
    period: '2008',
    latestEnglandHealthData: healthDataPoint,
    unitLabel: '',
  },
  {
    indicatorId: 2,
    indicatorName: 'no data indicator',
    period: '',
    latestEnglandHealthData: undefined,
    unitLabel: '',
  },
];

const mockAreaName = 'mockName';

describe('BasicTable', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <BasicTable areaName={mockAreaName} indicatorData={mockIndicatorData} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render BasicTable component', () => {
    render(
      <BasicTable areaName={mockAreaName} indicatorData={mockIndicatorData} />
    );
    expect(screen.getByRole('table')).toBeInTheDocument();
    expect(screen.getByTestId('basicTable-component')).toBeInTheDocument();
  });

  it('should always display indicator name when no health data for the indicator is present', () => {
    render(
      <BasicTable areaName={mockAreaName} indicatorData={mockIndicatorData} />
    );

    expect(screen.getAllByRole('row')[3]).toHaveTextContent(
      'no data indicator'
    );
    expect(screen.getAllByRole('cell')[3]).toHaveTextContent('X');
  });

  it('should display X in the table cell if there is no value', () => {
    render(
      <BasicTable areaName={mockAreaName} indicatorData={mockIndicatorData} />
    );

    const noValueCells = screen.getAllByText('X');
    expect(noValueCells).toHaveLength(5);
  });

  it('should display the correct aria label when then is no value', () => {
    render(
      <BasicTable areaName={mockAreaName} indicatorData={mockIndicatorData} />
    );

    const noValueCells = screen.getAllByLabelText('Not compared');
    expect(noValueCells).toHaveLength(5);
  });

  it('should render the correct trend label for the data in the table', () => {
    render(
      <BasicTable areaName={mockAreaName} indicatorData={mockIndicatorData} />
    );

    const trendTags = screen.getAllByTestId('trend-tag-component');
    expect(trendTags).toHaveLength(2);
    expect(trendTags[0].textContent).toEqual('No recent trend data available');
    expect(trendTags[1].textContent).toEqual('No recent trend data available');
  });
});
