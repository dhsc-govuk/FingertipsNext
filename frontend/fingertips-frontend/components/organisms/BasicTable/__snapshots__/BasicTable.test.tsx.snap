// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTable should match snapshot 1`] = `
<DocumentFragment>
  .c9 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 19px;
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  width: 100%;
  margin-top: 0;
  padding: 7px 10px;
  border: 2px solid transparent;
  border-radius: 0;
  color: #ffffff;
  background-color: #00703c;
  box-shadow: 0 2px 0 #002413;
  text-align: center;
  vertical-align: top;
  cursor: pointer;
  -webkit-appearance: none;
  margin-bottom: 22px;
}

.c9:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c9:link,
.c9:visited,
.c9:active,
.c9:hover {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9::-moz-focus-inner {
  padding: 0;
  border: 0;
}

.c9:hover,
.c9:focus {
  background-color: #00572e;
}

.c9:active {
  top: 2px;
  box-shadow: none;
}

.c9::before {
  content: "";
  display: block;
  position: absolute;
  top: -2px;
  right: -2px;
  bottom: -4px;
  left: -2px;
  background: transparent;
}

.c9:active::before {
  top: -4px;
}

.c9:disabled {
  opacity: 0.5;
  background: #00703c;
}

.c9:disabled:hover {
  background-color: #00703c;
  cursor: default;
}

.c9:disabled:focus {
  outline: none;
}

.c9:disabled:active {
  top: 0;
  box-shadow: 0 2px 0 #002413;
}

.c9 svg {
  max-width: 15px;
}

.c7 {
  margin: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 15px;
}

.c7 > p {
  margin: 0;
}

.c7 > p > code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.c7 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c7 > pre > code {
  display: inline;
  padding: 0;
  margin: 0;
  border: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 14px;
  line-height: 1;
  display: inline-block;
  padding-top: 5px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background-color: #1d70b8;
  color: #ffffff;
}

.c1 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c1:last-child {
  padding-right: 0;
}

.c2 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
}

.c2:last-child {
  padding-right: 0;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  text-transform: unset;
  margin: 0.3125em;
  color: #303436;
  background-color: #E1E2E3;
}

.c8 {
  margin-bottom: 0;
  margin-left: 0.3125em;
  position: relative;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
}

.c3 {
  text-align: center;
  padding-left: 10px;
  padding-right: 10px;
}

.c10 {
  margin-top: 1rem;
}

@media print {
  .c9 {
    font-size: 14px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c9 {
    font-size: 19px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c9 {
    width: auto;
  }
}

@media only screen and (min-width:641px) {
  .c9 svg {
    margin-left: 10px;
  }
}

@media only screen and (min-width:641px) {
  .c9 {
    margin-bottom: 32px;
  }
}

@media print {
  .c7 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    margin-bottom: 20px;
  }
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 16px;
    line-height: 1;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

@media print {
  .c8 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    font-size: 16px;
    line-height: 1.25;
  }
}

<div
    data-testid="basicTable-component"
  >
    <table
      class="c0"
      id="basicTable"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class="c1"
            colspan="6"
            style="padding-left: 10px;"
          >
            mockName
          </th>
        </tr>
        <tr
          class=""
        >
          <th
            class="c1"
            style="vertical-align: top; padding-left: 10px;"
          >
            Indicator
          </th>
          <th
            class="c1"
            style="vertical-align: top;"
          >
            Period
          </th>
          <th
            class="c1"
            style="vertical-align: top; text-align: right;"
          >
            Count
          </th>
          <th
            class="c1"
            style="vertical-align: top; text-align: left;"
          >
            Value unit
          </th>
          <th
            class="c1"
            style="vertical-align: top; text-align: right;"
          >
            Value
          </th>
          <th
            class="c1"
            style="vertical-align: top; text-align: center;"
          >
            Recent trend
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class="c2"
            style="padding-left: 10px;"
          >
             
          </td>
          <td
            class="c2"
            style="text-align: right;"
          >
            2008
          </td>
          <td
            class="c2"
            style="text-align: right;"
          >
            389
          </td>
          <td
            aria-label="Not compared"
            class="c2"
            style="text-align: left;"
          >
            X
          </td>
          <td
            class="c2"
            style="text-align: right;"
          >
            278.3
          </td>
          <td
            class="c2 c3"
          >
            <div
              data-testid="trendTag-container"
            >
              <strong
                class="c4 c5"
              >
                <div
                  class="c6"
                  data-testid="trend-tag-component"
                >
                  <div>
                    <div
                      class="c7 c8"
                    >
                      <p>
                        No recent trend data available
                      </p>
                    </div>
                  </div>
                </div>
              </strong>
            </div>
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="c2"
            style="padding-left: 10px;"
          >
            no data indicator
          </td>
          <td
            aria-label="Not compared"
            class="c2"
            style="text-align: right;"
          >
            X
          </td>
          <td
            aria-label="Not compared"
            class="c2"
            style="text-align: right;"
          >
            X
          </td>
          <td
            aria-label="Not compared"
            class="c2"
            style="text-align: left;"
          >
            X
          </td>
          <td
            aria-label="Not compared"
            class="c2"
            style="text-align: right;"
          >
            X
          </td>
          <td
            class="c2 c3"
          >
            <div
              data-testid="trendTag-container"
            >
              <strong
                class="c4 c5"
              >
                <div
                  class="c6"
                  data-testid="trend-tag-component"
                >
                  <div>
                    <div
                      class="c7 c8"
                    >
                      <p>
                        No recent trend data available
                      </p>
                    </div>
                  </div>
                </div>
              </strong>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      class="c9 c10"
    >
      Export options
    </button>
  </div>
</DocumentFragment>
`;
